lms/djangoapps/grades/api/v1/views.py

```diff
@@ -0,0 +1,233 @@
+                developer_message='The provided course key cannot be parsed.',
+                error_code='invalid_course_key'
+            )
+
+        try:
+            return courses.get_course_with_access(
+                user,
+                access_action,
+                course_key,
+                check_if_enrolled=True,
+            )
+        except Http404:
+            log.info('Course with ID "%s" not found', course_key_string)
+        except CourseAccessRedirect:
+            log.info('User %s does not have access to course with ID "%s"', user.username, course_key_string)
+        return self.make_error_response(
+            status_code=status.HTTP_404_NOT_FOUND,
+            developer_message='The user, the course or both do not exist.',
+            error_code='user_or_course_does_not_exist',
+        )
+
+    def _get_effective_user(self, request, course):
+        """
+        Returns the user object corresponding to the request's 'username' parameter,
+        or the current request.user if no 'username' was provided.
+
+        Verifies that the request.user has access to the requested users's grades.
+        Returns a 403 error response if access is denied, or a 404 error response if the user does not exist.
+        """
+
+        # Use the request user's if none provided.
+        if 'username' in request.GET:
+            username = request.GET.get('username')
+        else:
+            username = request.user.username
+
+        if request.user.username == username:
+            # Any user may request her own grades
+            return request.user
+
+        # Only a user with staff access may request grades for a user other than herself.
+        if not has_access(request.user, CourseStaffRole.ROLE, course):
+            log.info(
+                'User %s tried to access the grade for user %s.',
+                request.user.username,
+                username
+            )
+            return self.make_error_response(
+                status_code=status.HTTP_403_FORBIDDEN,
+                developer_message='The user requested does not match the logged in user.',
+                error_code='user_mismatch'
+            )
+
+        try:
+            return USER_MODEL.objects.get(username=username)
+
+        except USER_MODEL.DoesNotExist:
+            return self.make_error_response(
+                status_code=status.HTTP_404_NOT_FOUND,
+                developer_message='The user matching the requested username does not exist.',
+                error_code='user_does_not_exist'
+            )
+
+    def _make_grade_response(self, user, course, course_grade, use_email=False):
+        """
+        Serialize a single grade to dict to use in Repsonses
+        """
+        if use_email:
+            user = user.email
+        else:
+            user = user.username
+
+        return {
+            'user': user,
+            'course_key': str(course.id),
+            'passed': course_grade.passed,
```

Comment:
This will generate a `NullPointer` exception when `course_grade` is `None` in case of error strings returned by `CourseGradeFactory.iter`.  See comment below.